### PR TITLE
Prevent creating databases in Vitess until it works better

### PR DESCRIPTION
**Pull request by pimeys**

### CODE REVIEW

**Review by tomhoule — APPROVED** [review]
Not too happy about the extra roundtrip, but let's merge this, maybe it won't need to stay long, with what they are working on on the vitess side

**Review by janpio — APPROVED** [review]

File: migration-engine/connectors/sql-migration-connector/src/flavour/mysql.rs [hunk 1]
```diff
@@ -221,10 +227,24 @@ impl SqlFlavour for MysqlFlavour {
...
-        let version = connection.version().await?;
+        let version = connection
+            .query_raw("SELECT @@version", &[])
+            .await?
+            .into_iter()
+            .next()
+            .and_then(|r| r.into_iter().next())
+            .and_then(|val| val.into_string());
+
+        let global_version = connection.version().await?;
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/flavour/mysql.rs:238]: If we want to micro-optimize, we could even fetch the two version strings in one query :p

**Review by tomhoule — COMMENTED** [review]
LGTM!

File: migration-engine/connectors/sql-migration-connector/src/flavour/mysql.rs (hunk 1 above)
Comment (pimeys) [reply migration-engine/connectors/sql-migration-connector/src/flavour/mysql.rs:238]: I'd like to not change Quaint APIs for this now though.

Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/flavour/mysql.rs:238]: I meant by copying the query quaint does (probably `@@global.version`, so `SELECT @@version, @@global.version`)

**Pull request merged**